### PR TITLE
Enable TLS between Ironic and Mariadb

### PR DESCRIPTION
**Pull request by namnx228**
This PR is related to:
https://github.com/metal3-io/ironic-image/pull/206 
https://github.com/metal3-io/baremetal-operator/pull/689
The two PRs above need to be merged before this PR can run the CI tests

### CODE REVIEW

File: lib/ironic_tls_setup.sh [hunk 1]
```diff
@@ -31,6 +39,9 @@ if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
     if [ ! -f "${IRONIC_INSPECTOR_CACERT_FILE}" ]; then
         openssl req -x509 -new -nodes -key "${IRONIC_INSPECTOR_CAKEY_FILE}" -sha256 -days 1825 -out "${IRONIC_INSPECTOR_CACERT_FILE}" -subj /CN="ironic inspector CA"/
     fi
+    if [ ! -f "${MARIADB_CACERT_FILE}" ]; then
```
Comment (maelk) [inline lib/ironic_tls_setup.sh:42]: You're missing the CAKEY generation for mariadb

**Review by maelk — CHANGES_REQUESTED** [review]

File: lib/ironic_tls_setup.sh [hunk 2]
```diff
@@ -73,4 +92,7 @@ else
     unset IRONIC_INSPECTOR_CACERT_FILE
     unset IRONIC_INSPECTOR_CERT_FILE
     unset IRONIC_INSPECTOR_KEY_FILE
-fi
\ No newline at end of file
+    unset MARIADB_CACERT_FILE
+    unset MARIADB_CERT_FILE
+    unset MARIADB_KEY_FILE
```
Comment (fmuyassarov) [inline lib/ironic_tls_setup.sh:97]: Would you mind also to update the vars.md with these new vars?

**Review by furkatgofurov7 — COMMENTED** [review]
This needs rebase, otherwise looks good once comments addressed

File: lib/ironic_tls_setup.sh (hunk 1 above)
Comment (namnx228) [reply lib/ironic_tls_setup.sh:45]: Fixed!. Thanks

File: lib/ironic_tls_setup.sh (hunk 2 above)
Comment (namnx228) [reply lib/ironic_tls_setup.sh:100]: I have added that. Thanks

**Review by maelk — APPROVED** [review]

File: 03_launch_mgmt_cluster.sh [hunk 3]
```diff
@@ -204,6 +207,7 @@ IRONIC_ENDPOINT=${IRONIC_URL}
 IRONIC_INSPECTOR_ENDPOINT=${IRONIC_INSPECTOR_URL}
 CACHEURL=http://$IRONIC_HOST/images
 IRONIC_FAST_TRACK=true
+IRONIC_TLS_SETUP=${IRONIC_TLS_SETUP}
```
Comment (furkatgofurov7) [inline 03_launch_mgmt_cluster.sh:210]: why do we need this here?

**Review by furkatgofurov7 — COMMENTED** [review]
Some questions inline

File: lib/ironic_tls_setup.sh [hunk 4]
```diff
@@ -16,13 +16,24 @@ if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
     export IRONIC_INSPECTOR_CERT_FILE="${IRONIC_INSPECTOR_CERT_FILE:-"${WORKING_DIR}/certs/ironic-inspector.crt"}"
     export IRONIC_INSPECTOR_KEY_FILE="${IRONIC_INSPECTOR_KEY_FILE:-"${WORKING_DIR}/certs/ironic-inspector.key"}"
 
+    export MARIADB_CACERT_FILE="${MARIADB_CACERT_FILE:-"${WORKING_DIR}/certs/ironic-ca.pem"}"
+    export MARIADB_CAKEY_FILE="${IRONIC_CAKEY_FILE:-"${WORKING_DIR}/certs/ironic-ca.key"}"
+    export MARIADB_CERT_FILE="${MARIADB_CERT_FILE:-"${WORKING_DIR}/certs/mariadb.crt"}"
+    export MARIADB_KEY_FILE="${MARIADB_KEY_FILE:-"${WORKING_DIR}/certs/mariadb.key"}"
+    MARIADB_HOST="${MARIADB_HOST:-"mariadb"}"
+    MARIADB_HOST_IP="${MARIADB_HOST_IP:-"127.0.0.1"}"
+
```
[the remark this answers is not in the document]
Comment (namnx228) [reply lib/ironic_tls_setup.sh:23]: That is true. We don't need them. Fixed!!!

File: 03_launch_mgmt_cluster.sh (hunk 3 above)
Comment (namnx228) [reply 03_launch_mgmt_cluster.sh]: Removed! Thanks for your review

**Comment by maelk** [comment]
you'll impact the pivoting and upgrade tests with this PR, so you need to run the tests once you have the ironic image and BMO PR in

File: 03_launch_mgmt_cluster.sh [hunk 5]
```diff
@@ -12,6 +12,8 @@ source lib/network.sh
 
 export IRONIC_HOST="${CLUSTER_URL_HOST}"
 export IRONIC_HOST_IP="${CLUSTER_PROVISIONING_IP}"
+export MARIADB_HOST="mariaDB"
+export MARIADB_HOST_IP="127.0.0.1"
```
Comment (fmuyassarov) [inline 03_launch_mgmt_cluster.sh:16]: nit: Maybe these could be parameterized in lib/common.sh?

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Pull request merged**